core/common/src/ByteStrings.kt

```diff
@@ -85,6 +85,8 @@ public fun Source.readByteString(byteCount: Int): ByteString {
  * expands the source's buffer as necessary until [byteString] is found. This reads an unbounded number of
  * bytes into the buffer. Returns `-1` if the stream is exhausted before the requested bytes are found.
  *
+ * For empty byte strings this function always return `0`.
```

Comment:
See #423. I'm not sure how to handle empty byte strings yet.

Reply:
It seems to me that this is really better than fetching the data, provided that we always ignore what has been fetched.

Maybe it should be explicitly written that in the case of an empty byte string, the data will never be read from the source?

Reply:
> Maybe it should be explicitly written that in the case of an empty byte string, the data will never be read from the source?

Then, it would be inconsistent with the `Buffer`'s "overload". These two functions already behave differently, but I'd like to keep them as close to each other as possible.